libgoal/participation.go

```diff
@@ -157,7 +157,11 @@ func (c *Client) GenParticipationKeysTo(address string, firstValid, lastValid, k
 	}
 	_, err = os.Stat(partKeyPath)
 	if !os.IsNotExist(err) {
-		err = fmt.Errorf("ParticipationKeys exist for the range %d to %d", firstRound, lastRound)
+		if err != nil {
```

Comment:
If find this flow a bit confusing to read, how about rewriting it as:

```
if err == nil {
	err = fmt.Errorf("ParticipationKeys exist for the range %d to %d", firstRound, lastRound)
	return
} else if !os.IsNotExist(err) {
	err = fmt.Errorf("participation key file '%s' cannot be accessed : %w", partKeyPath, err)
	return
}
```

Reply:
done.